.github/workflows/pkgci_sharktank_regression.yml

```diff
@@ -0,0 +1,174 @@
+        type: string
+        default: ""
+
+jobs:
+  sharktank_regression_tests:
+    name: "sharktank_regression_tests :: ${{ matrix.name }}"
+    runs-on: ${{ matrix.runs-on }}
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          # CPU
+          - name: cpu_task_model
+            target: target_cpu
+            gpu: none
+            runs-on: ubuntu-24.04
+          - name: cpu_llvm_task_regression
+            models-config-file: models_cpu_llvm_task.json
+            backend: cpu
+            iree_test_files: /home/nod/iree_tests_cache
+            sku: cpu
+            runs-on:
+              - self-hosted # must come first
+              - persistent-cache
+              - Linux
+              - X64
+
+          # AMD GPU
+          - name: rocm_hip_w7900_model
+            target: target_hip
+            gpu: gfx1100
+            runs-on: nodai-amdgpu-w7900-x86-64
+          - name: rocm_hip_mi300_model
+            target: target_hip
+            gpu: gfx942
+            runs-on: linux-mi300-2gpu-ossci-iree-org
+          - name: amdgpu_rocm_mi250_gfx90a_regression
+            rocm-chip: gfx90a
+            backend: rocm
+            iree_test_files: /home/esaimana/iree_tests_cache
+            sku: mi250
+            runs-on: nodai-amdgpu-mi250-x86-64
+          - name: amdgpu_rocm_mi300_gfx942_regression
+            rocm-chip: gfx942
+            backend: rocm
+            iree_test_files: /home/esaimana/iree_tests_cache
+            sku: mi300
+            runs-on: linux-mi300-1gpu-ossci-iree-org
+          - name: amdgpu_rocm_mi308_gfx942_regression
+            rocm-chip: gfx942
+            backend: rocm
+            iree_test_files: /home/esaimana/iree_tests_cache
+            sku: mi308
+            runs-on: nodai-amdgpu-mi308-x86-64
+
+    env:
+      VENV_DIR: ${{ github.workspace }}/venv
+      GH_TOKEN: ${{ github.token }}
+      PACKAGE_DOWNLOAD_DIR: ${{ github.workspace }}/.packages
+      TEST_OUTPUT_ARTIFACTS: ${{ github.workspace }}/model_output_artifacts
+      IREE_TEST_FILES: ${{ matrix.iree_test_files }}
+    steps:
+      - name: Checking out IREE repository
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          submodules: false
+
+      - uses: actions/setup-python@8d9ed9ac5c53483de85588cdf95a591a75ab9f55 # v5.4.0
+        with:
+          # Must match the subset of versions built in pkgci_build_packages.
+          python-version: "3.11"
+
+      - uses: actions/download-artifact@95815c38cf2ff2164869cbab79da8d1f422bc89e # v4.2.1
+        if: ${{ inputs.artifact_run_id == '' }}
+        with:
+          name: linux_x86_64_release_packages
+          path: ${{ env.PACKAGE_DOWNLOAD_DIR }}
+
+      - name: "Set IREE_TEST_FILES variable"
+        run: |
+          case "${{ matrix.name }}" in
+            "amdgpu_rocm_mi300_gfx942_regression") echo IREE_TEST_FILES="/shark-cache/data/iree-regression-cache" >> $GITHUB_ENV ;;
```

Comment:
Some of this code was preexisting from @Eliasj42 , but I think we should actually check `runs-on: linux-mi300-1gpu-ossci-iree-org` here, not the name. The _runner_ has access to a cache there, so the check should be specific. We could check for `contains` with `ossci` if we want to generalize to `2gpu`, `mi325`, and other runner labels that may be added in the future, assuming they have access to the same cache directory path.